mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -194,6 +194,41 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	c.clientEventarcDCL = eventarcDcl.NewClient(c.dclConfig)
 <% end -%>
 
+	// Print Identities executing terraform API Calls.
+
+	if c.ImpersonateServiceAccount != "" {
+		// Drop Impersonated ClientOption from OAuth2 TokenSource to infer original identity
+
+		tokenSource, err = c.getTokenSource(c.Scopes, true)
```

Comment:
Can we define some named constants and pass those in instead of raw booleans? For example `WithImpersonation` and `InitialCredentialsOnly`

Reply:
I don't follow but I did a comment to the function so godoc appears with details when you hover over getTokenSource